### ISSUE TITLE
[AsciiDoc conversion] Cross reference text

### ISSUE THREAD

**Issue by iansk** [hlxsites/prisma-cloud-docs-website#137]
If left unspecified, the link text for cross references should automatically be derived from the section heading. That default AsciiDoctor behavior isn't working on Franklin. The link text currently appears as "undefined".

**AsciiDoctor input:**

https://github.com/hlxsites/prisma-cloud-docs/blob/8035541314facb205bd33ac6761bcdeb270306f4/docs/en/compute/pcee/admin-guide/welcome/test-asciidoc.adoc?plain=1#L47-L50

```
The following links take you to nested subsections.

* <<test-xref-to-anchor>>
* <<test-xref-to-named-anchor>>
```

**Franklin output:**

https://main--prisma-cloud-docs-website--hlxsites.hlx.page/prisma/prisma-cloud/en/compute/pcee/admin-guide/welcome/test-asciidoc?branch=ian/test-asciidoc

Scroll down to the section "Internal cross references". The attached screenshot shows what gets rendered on the page versus what should actually be rendered on the page.

<img width="981" alt="franklin-uat-cross-ref-text" src="https://github.com/hlxsites/prisma-cloud-docs-website/assets/6518946/c00abb31-0cb9-4fe5-b146-5b6b3013a3a1">

**Reference info:**

* https://docs.asciidoctor.org/asciidoc/latest/macros/xref/#internal-cross-references

**Comment by maxakuru** [comment]
@iansk we don't have the title of the referenced document at conversion time, and finding it would require 1 additional requests for each xref

I suggest we avoid doing that, and instead just require xrefs to be titled during authoring.. wdyt? is that reasonable?

**Comment by iansk** [comment]
@maxakuru Wherever we can, I'd like our Franklin implementation to support AsciiDoctor's standard, documented behavior.

Can we use the same logic as the standard AsciiDoctor HTML5 converter? See the source [here](https://github.com/asciidoctor/asciidoctor/blob/454d590906459dc4645f9b01e44df486c1385827/lib/asciidoctor/converter/html5.rb#L1116-L1139)

As I understand the code, if you've got a cross-doc reference, you need to use the `xref:PATH[TEXT]` macro, where the link text (`TEXT`) must be explicitly specified, and in code you can grab the link text with `node.getText()`.

For same-doc references, AsciiDoc has a shorthand syntax `<<ANCHOR>>`. If you use the shorthand syntax, you don't need to specify the link text because AsciiDoc keeps a "reference catalog" for the document being converted. Using the node's `refid` you can look up the `xreftext` in the reference catalog.

**Reference info:**

* https://docs.asciidoctor.org/asciidoc/latest/macros/xref/
* https://docs.asciidoctor.org/pdf-converter/latest/interdocument-xrefs/

**Comment by maxakuru** [comment]
@iansk for inline references you're correct - I was referring to external docs in the previous comment. this is added/fixed in #143, https://github.com/hlxsites/prisma-cloud-docs/pull/71 and https://github.com/hlxsites/prisma-cloud-docs/pull/72

I'll leave this open for you to try it out

**Comment by iansk** [comment]
Fixed. Thanks @maxakuru
